File: Algorithm/Leetcode/0002.py
# 1108. Defanging an IP Address

"""https://leetcode.com/problems/defanging-an-ip-address/discuss/340369/Python-One-Line"""

class Solution:
    def defangIPaddr(self, address: str) -> str:
      return '[.]'.join(address.split('.'))

File: Algorithm/Leetcode/0002.cpp
// 1108. Defanging an IP Address

// https://leetcode.com/problems/defanging-an-ip-address/discuss/328855/C%2B%2B-1-liner-(regex_replace)

class Solution {
public:
    string defangIPaddr(string address) {
        return regex_replace(address, regex("[.]"), "[.]");
    }
};
